### ISSUE TITLE
False Positive: Lint/InterpolationCheck

### ISSUE THREAD

**Issue by jfelchner** [bbatsov/rubocop#5357]
## Expected behavior

The cop should understand matching things that look like string interpolations inside a Regex

## Actual behavior

Matching `\#{20}` is matching 20 `#` in a row, not a string interpolation for a variable named `20`

## Steps to reproduce the problem

```ruby
# baz.rb
class Foo
  def bar(qux)
    /\#{20}/
  end
end
```

    > rubocop --only='Lint/InterpolationCheck' baz.rb

## Output

```
Inspecting 1 file
W

Offenses:

baz.rb:3:6: W: Lint/InterpolationCheck: Interpolation in single quoted string detected. Use double quoted strings if you need interpolation.
    /\#{20}/
     ^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.4.3 x86_64-darwin16)
```

**Comment by bbatsov** [comment]
I also notice that the message `Interpolation in single quoted string detected. Use double quoted strings if you need interpolation.` violates our policy to have a single-sentence cop messages. We should fix this as well.

**Comment by jamiemccarthy** [comment]
I'm seeing this too, also with an escaped `#` in a double-quoted string. My test case is:

## Steps to reproduce the problem

```
root@ac560a15cb0c:/app# cat interp.rb
# frozen_string_literal: true

class Interp # :nodoc:
  my_string = "\#{msg}"
  puts my_string
end
root@ac560a15cb0c:/app# rubocop interp.rb
```

## Output

```
Inspecting 1 file
W

Offenses:

interp.rb:4:15: W: Lint/InterpolationCheck: Interpolation in single quoted string detected. Use double quoted strings if you need interpolation.
  my_string = "\#{msg}"
              ^^^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
root@ac560a15cb0c:/app# rubocop -v
0.52.1
```
